txp/Pos/StateLock/PrioLock.hs

```diff
@@ -0,0 +1,77 @@
+{-|
+Module:      Pos.StateLock.PrioLock
+Description: Provides a prioritised lock
+
+Provides a lock that can be taken with either high or low precedence.
+Within each precedence, the lock is taken in FIFO order.
+-}
+
+{-# LANGUAGE LambdaCase #-}
+
+module Pos.StateLock.PrioLock where
```

Comment:
Limit the export list to the public API. That may just be `PriorityLock`, `newPriorityLock`, `Priority` and `withPriorityLock`.

Reply:
Thanks, I'll do that!